common/stringFunctions.lua

```diff
@@ -182,3 +182,66 @@ if not string.formatSI then
 		return str .. siPrefix
 	end
 end
+
+if not string.Levenshtein then
+	-- reuseable tables
+	local Levenshtein0 = {}
+	local Levenshtein1 = {}
+	local min = math.min
+	local ssub = string.sub
+	
+	--- Calculates edit distance of two strings, O(n*m) time, O(n) memory
+	---@param a string
+	---@param b string
+	---@return number 
+
+	function string.Levenshtein(a,b)
+		local lena = string.len(a)
+		local lenb = string.len(b)
+		local ssub = string.sub
+		
+		Levenshtein1[1] = 0
+		for c = 0, lenb do -- initialize the first row
+			Levenshtein0[c+1] = c
+		end
+		for r = 1, lena do
+			--print(table.unpack(Levenshtein0))
+			for c = 0, lenb do -- 16 ns/loop wtf
+				if c == 0 then
+					Levenshtein1[1] = r
+				else
+					Levenshtein1[c+1] = min(
+						min(Levenshtein0[c+1] + 1,	Levenshtein1[c] + 1),
+						Levenshtein0[c] + (ssub(a,r,r) == ssub(b,c,c) and 0 or 1)
+					)
+				end
+			end
+			Levenshtein0, Levenshtein1 = Levenshtein1, Levenshtein0 -- swap rows
+		end
+		return Levenshtein1[lenb]
```

Comment:
The function gives incorrect output for at least these test cases:
```lua
{
	name = "returns the length of the non-empty string for one empty string",
	args = { "", "apple" },
	expected = 5,
},
{
	name = "returns the length of the non-empty string for one empty string (reverse case)",
	args = { "apple", "" },
	expected = 5,
},
{
	name = "returns 1 for strings differing by one character",
	args = { "apple", "appla" },
	expected = 1,
},
{
	name = "returns the length of the longer string for completely different strings",
	args = { "cat", "dogs" },
	expected = 4,
},
```
This change fixes those tests:
```suggestion
		return Levenshtein0[lenb + 1]
```